resolwe/flow/managers/listener/python_process_plugin.py

```diff
@@ -99,6 +99,30 @@ class ExposeObjectPlugin(metaclass=abc.ABCMeta):
 
     full_model_name = "app_label.model_name"
 
+    def _has_permission(
+        self, user: UserClass, model: Type[Model], model_pk: int, permission_name: str
+    ):
+        """Check if contributor has requested permissions.
+
+        :raises RuntimeError: with detailed explanation when check fails.
+        """
+        full_permission_name = get_full_perm(permission_name, model)
+        object_ = model.objects.filter(pk=model_pk)
+        filtered_object = get_objects_for_user(
+            user,
+            [full_permission_name],
+            object_,
+        )
+        if not filtered_object:
+            if object_:
+                raise RuntimeError(
+                    f"No permissions: {model._meta.model_name} with id {model_pk}."
+                )
+            else:
+                raise RuntimeError(
```

Comment:
You can raise this exception before checking permissions.

Reply:
OK, more efficient. I will change that and add tests.